### PR TITLE
drivers/sdio: Call SDIO_LOCK before and after the transaction

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
follow the same behaviour in drivers/mmcsd/mmcsd_sdio.c

## Impact
sdio driver

## Testing
Pass CI

### CODE REVIEW

File: drivers/mmcsd/sdio.c [hunk 1]
```diff
@@ -115,6 +124,10 @@ int sdio_sendcmdpoll(FAR struct sdio_dev_s *dev, uint32_t cmd, uint32_t arg)
   return ret;
 }
 
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
 int sdio_io_rw_direct(FAR struct sdio_dev_s *dev, bool write,
                       uint8_t function, uint32_t address,
                       uint8_t inb, uint8_t *outb)
```
Comment (pkarashchenko) [inline drivers/mmcsd/sdio.c:133]:
```suggestion
                      uint8_t inb, FAR uint8_t *outb)
```

please add `FAR` to `uint8_t *buf` in `sdio_io_rw_extended` and to pointer in `sdio_set_wide_bus`

File: include/nuttx/sdio.h [hunk 2]
```diff
@@ -1044,9 +1044,6 @@ int sdio_enable_function(FAR struct sdio_dev_s *dev, uint8_t function);
 
 int sdio_enable_interrupt(FAR struct sdio_dev_s *dev, uint8_t function);
 
-int sdio_sendcmdpoll(FAR struct sdio_dev_s *dev,
-                     uint32_t cmd, uint32_t arg);
-
 int sdio_io_rw_direct(FAR struct sdio_dev_s *dev, bool write,
                       uint8_t function, uint32_t address,
                       uint8_t inb, uint8_t *outb);
```
Comment (pkarashchenko) [inline include/nuttx/sdio.h:1049]:
```suggestion
                      uint8_t inb, FAR uint8_t *outb);
```
and add `FAR` to `uint8_t *buf` in `sdio_io_rw_extended`.

File: drivers/mmcsd/sdio.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/mmcsd/sdio.c]: Done.

File: include/nuttx/sdio.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/nuttx/sdio.h]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**